rules/rules.go

```diff
@@ -26,9 +28,20 @@ func (r *Rules) host(hosts ...string) *mux.Route {
 		if err != nil {
 			reqHost = req.Host
 		}
-		for _, host := range hosts {
-			if types.CanonicalDomain(reqHost) == types.CanonicalDomain(host) {
-				return true
+		if r.HostResolver != nil && r.HostResolver.Enabled {
+			reqH, flatH := r.HostResolver.CNAMEFlatten(types.CanonicalDomain(reqHost))
+			for _, host := range hosts {
+				if types.CanonicalDomain(reqH) == types.CanonicalDomain(host) {
+					return true
+				} else if types.CanonicalDomain(flatH) == types.CanonicalDomain(host) {
```

Comment:
Can you remove the `else` statement which is not mandatory after a `return` please?
Only use another `if` statement.